hecuba_py/src/hfilter.py

```diff
@@ -0,0 +1,241 @@
+    func_vars, clean_string = func_to_str(func)
+    magical_regex = regex.compile('(?:\d+(?:\.\d+)?|\w|"\w+")+|[^\s\w\_]')
+    parsed_string = magical_regex.findall(clean_string)
+    # Fusing .'s, symbols
+    # print(str(parsed_string))
+    for i, elem in enumerate(parsed_string):
+        #if elem.find('.') > 0  and len(elem) is 1:
+        try:
+            if elem.index('.') > -1:
+                parsed_string[i - 1:i + 2] = [''.join(parsed_string[i - 1:i + 2])]
+        except:
+            if (elem is '=') and re.match('(>|<)', parsed_string[i - 1]) is not None:
+                parsed_string[i - 1:i + 1] = [''.join(parsed_string[i - 1:i + 1])]
+            elif (elem is '=') and (parsed_string[i - 1] is '='):
+                parsed_string[i - 1:i + 1] = '='
+
+    # Getting variables
+    dictv = {}
+    for i, elem in enumerate(func.__code__.co_freevars):
+        dictv[str(elem)] = func.__closure__[i].cell_contents
+
+    # Combine set or tuple
+    for i, elem in enumerate(parsed_string):
+        if elem is "[":
+            index = parsed_string[i:].index(']')
+            c = ''.join(parsed_string[i:index + i + 1])
+            parsed_string[i:index + i + 1] = [eval(c)]
+        elif elem is '(':
+            index = parsed_string[i:].index(')')
+            c = ''.join(parsed_string[i:index + i + 1])
+            parsed_string[i:index + i + 1] = [eval(c)]
+
+    # Creating sublists
+    lastpos = 0
+    newpos = 0
+    final_list = []
+    if len(parsed_string) > 3:
+        while newpos < len(parsed_string):
+            if 'and' in parsed_string[lastpos:]:
+                newpos = parsed_string[lastpos:].index('and')
+                newpos = newpos + lastpos
+            else:
+                newpos = len(parsed_string)
+            sublist = parsed_string[lastpos:newpos]
+            lastpos = newpos + 1
+            sublist = substit_var(sublist, func_vars, dictv)
+
+            final_list.append(sublist)
+    else:
+
+        sublist = substit_var(parsed_string, func_vars, dictv)
+        final_list.append(sublist)
+    # Replace types for correct ones
+
+    final_list = transform_to_correct_type(final_list, dictv)
+    # print(final_list)
+    return final_list
+
+
+def hfilter(lambda_filter, iterable):
+    if not isinstance(iterable, IStorage):
+        return python_filter(lambda_filter, iterable)
+
+    if not iterable._is_persistent:
+        raise Exception("The StorageDict needs to be persistent.")
+
+    parsed_lambda = parse_lambda(lambda_filter)
+    predicate = Predicate(iterable)
+    for expression in parsed_lambda:
+        if expression[1] in (">", "<", "=", ">=", "<="):
+            predicate = predicate.comp(col=expression[0], comp=expression[1], value=expression[2])
+        elif expression[1] == "in":
+            predicate = predicate.inside(col=expression[0], values=expression[2])
+        else:
+            raise Exception("Bad expression.")
+
+    return predicate.execute()
```

Comment:
A Python filter must return a List, you can't return a `cassandra.cluster.ResultSet` (I'd say iterate over the result set and return a list)